packages/common/src/api/library.ts

```diff
@@ -9,66 +9,96 @@ import { reformatCollection } from 'store/cache/collections/utils/reformatCollec
 import { encodeHashId } from 'utils/hashIds'
 import { removeNullable } from 'utils/typeUtils'
 
-type GetLibraryAlbumsArgs = {
+type GetLibraryItemsArgs = {
   userId: number
   offset: number
   limit: number
   query?: string
   sortMethod?: full.GetUserLibraryAlbumsSortMethodEnum
   sortDirection?: full.GetUserLibraryAlbumsSortDirectionEnum
 }
+const COLLECTIONS_CACHE_OPTIONS = {
+  kind: Kind.COLLECTIONS,
+  schemaKey: 'collections',
+  idListArgKey: 'playlist_id'
+}
+
+const fetchLibraryCollections = async ({
+  args,
+  context,
+  collectionType
+}: {
+  args: GetLibraryItemsArgs
+  context: AudiusQueryContextType
+  collectionType: 'album' | 'playlist'
+}) => {
+  const { audiusSdk, audiusBackend } = context
+  const {
+    userId,
+    offset,
+    limit,
+    query = '',
+    sortMethod = 'added_date',
+    sortDirection = 'desc'
+  } = args
+  const sdk = await audiusSdk()
+  const { data, signature } = await audiusBackend.signDiscoveryNodeRequest()
+
+  const getCollectionsParams = {
+    id: encodeHashId(userId),
+    userId: encodeHashId(userId),
+    offset,
+    limit,
+    query,
+    sortMethod,
+    sortDirection,
+    type: 'all' as full.GetUserLibraryAlbumsTypeEnum,
+    encodedDataMessage: data,
+    encodedDataSignature: signature
+  }
+  const { data: rawCollections = [] } =
+    collectionType === 'album'
+      ? await sdk.full.users.getUserLibraryAlbums(getCollectionsParams)
+      : await sdk.full.users.getUserLibraryPlaylists(getCollectionsParams)
+  const collectionsMetadata = rawCollections
+    .map((r: APIActivityV2) => makeActivity(r))
+    .filter(removeNullable) as UserCollectionMetadata[]
+  const collections = collectionsMetadata.map((am) =>
+    reformatCollection({
```

Comment:
FYI need to incorporate suggestion to move this from last PR still!

Reply:
Awesome! Should be easy, just copy the pattern in createApi for `reformatUser`